### PR TITLE
[mesheryctl] Filter: List subcommand dysfunctional

### LINKED ISSUE #7994: [mesheryctl] Filter: List subcommand dysfunctional
_See [mesheryctl Command Tracker](https://bit.ly/3dqXy1q) for current status of commands._

#### Current Behavior
```
$ mesheryctl exp filter list
Error: Error processing JSON response from server.
invalid character '<' looking for beginning of value
```

#### Desired Behavior
Returns a list of filters by name and ID

#### Environment

- Host OS: Mac 
- Meshery Server Version: stable-v
- Meshery Client Version: stable-v0.6.99 	33ece5d95

---

#### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 📚 [mesheryctl Contributing Guide](https://github.com/meshery/meshery/blob/master/mesheryctl/README.md)
- 📺 [Beginner's guide to contributing to Meshery and mesheryctl](https://youtu.be/hh_kFLZx3G4)
- ⌨️ [mesheryctl Command Tracker](https://docs.google.com/spreadsheets/d/1q63sIGAuCnIeDs8PeM-0BAkNj8BBgPUXhLbe1Y-318o/edit#gid=0)
- ⌨️ [Meshery CLI Commands and Documentation](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#heading=h.5fucij4hc5wt)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by suhail34**
**Notes for Reviewers**

This PR fixes #7994 

The error while using `mesheryctl exp filter list` was because the token get expires and when we make request `/api/filter` it redirect to login page so while unmarshalling in below code 
https://github.com/meshery/meshery/blob/80f6fea86ad2215e27f2028c7b5f22c8cf8d7b34/mesheryctl/internal/cli/root/filter/list.go#L70 

the response we get is not of type `models.FiltersAPIResponse` so that's the reason we get error Invalid character '<' looking for beginning of value ('<' this is an html open tag)

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

File: mesheryctl/internal/cli/root/filter/list.go [hunk 1]
```diff
@@ -54,7 +55,21 @@ mesheryctl exp filter list
...
+		}
+		jwtToken := map[string]string{}
+		errCheck := json.Unmarshal(file, &jwtToken)
+		if errCheck != nil {
+			return ErrUnmarshal(errCheck)
+		}
+		_, invalid := prov.VerifyToken(jwtToken["token"])
+		if invalid != nil {
+			return ErrInvalidAuthToken()
+		}
```
Comment (leecalcote) [inline mesheryctl/internal/cli/root/filter/list.go:72]: @alphaX86 @abdullah1308 @suhail34 where is our standard function for this that is reused across all commands?

**Review by leecalcote — CHANGES_REQUESTED** [review]

File: mesheryctl/internal/cli/root/filter/list.go (hunk 1 above)
Comment (suhail34) [reply mesheryctl/internal/cli/root/filter/list.go:72]: I see that when the token expires the same errors are with other subcommands too

Comment (suhail34) [reply mesheryctl/internal/cli/root/filter/list.go:72]: I didn't saw any standard function in mesheryctl folder that does verifytoken if there are please do point me

File: mesheryctl/internal/cli/root/filter/list.go [hunk 2]
```diff
@@ -54,7 +55,21 @@ mesheryctl exp filter list
...
 		res, err := client.Do(req)
```
Comment (abdullah1308) [inline mesheryctl/internal/cli/root/filter/list.go:73]: @suhail34 This logic is redundant as @leecalcote has pointed out. Use this function instead - 

https://github.com/meshery/meshery/blob/5c11aa7e6f1ff2c1cd3cb3f9990abfb05153a1f0/mesheryctl/pkg/utils/auth.go#L65-L94

Comment (abdullah1308) [inline mesheryctl/internal/cli/root/filter/list.go:73]: Replace with `utils.MakeRequest`

**Review by abdullah1308 — CHANGES_REQUESTED** [review]

**Comment by Chadha93** [comment]
@suhail34 Let's discuss this on the Meshery Dev call. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit#) if you would. :)

**Review by abdullah1308 — APPROVED** [review]

**Pull request merged**